plugins/kotlin/idea/src/org/jetbrains/kotlin/idea/quickfix/createFromUsage/createCallable/CreateCallableFromCallActionFactory.kt

```diff
@@ -140,8 +140,6 @@ sealed class CreateCallableFromCallActionFactory<E : KtExpression>(
     }
 
     protected fun getAbstractCallableInfo(mainCallable: CallableInfo, originalExpression: KtExpression): CallableInfo? {
-        if (mainCallable.isForCompanion) return null
-
```

Comment:
Reverted and added a test.

Reply:
`org.jetbrains.kotlin.idea.quickfix.QuickFixTestGenerated.CreateFromUsage.CreateFunction.Call.Abstract#testCompanion`  test doesn't fail anymore if you comment out this `if`.

It seems that after you migrated from `isForCompanion` to `staticContextRequired` this `if` isn't required anymore for some reason. Can you please elaborate why or cover this `if` with new tests?

Reply:
This `if` is not needed because `originalReceiverTypeInfo` becomes `TypeInfo.StaticContextRequired` and is returned here:

https://github.com/JetBrains/intellij-community/blob/aa444258814376186c42a5c41e57a6ff83320b49/plugins/kotlin/idea/src/org/jetbrains/kotlin/idea/quickfix/createFromUsage/createCallable/CreateCallableFromCallActionFactory.kt#L140

Reply:
I removed this `if`: aa44425